walletlibrary/src/main/java/com/microsoft/walletlibrary/util/VerifiedIdResult.kt

```diff
@@ -1,3 +1,58 @@
 package com.microsoft.walletlibrary.util
 
-typealias VerifiedIdResult<T> = Result<T>
\ No newline at end of file
+import com.microsoft.did.sdk.util.controlflow.NetworkException
+import com.microsoft.did.sdk.util.controlflow.SdkException
+
+typealias VerifiedIdResult<T> = Result<T>
+
+internal suspend fun <T> getResult(block: suspend () -> T): VerifiedIdResult<T> {
+    return try {
+        val result = block()
+        VerifiedIdResult.success(result)
+    } catch (verifiedIdException: VerifiedIdException) {
+        verifiedIdException.toVerifiedIdResult()
+    } catch (exception: WalletLibraryException) {
+        when (exception.cause) {
+            is NetworkException -> {
+                val networkException = exception.cause as NetworkException
```

Comment:
why does this need to be recast? Isn't it basically saying if NetworkException, rewrap in an NetworkException?

Reply:
What is the difference between WalletLibraryException and VerifiedIdException?

Reply:
VerifiedIdException is exposed externally and there is only a few of them specified in an enum. WalletLibraryException is internal exceptions which could in Wallet Library or VC SDK and are populated as innerErrors or cause in VerifiedIdException.